From 79db7bae0cf8e7eb5cf7c0958498de8f13a63852 Mon Sep 17 00:00:00 2001
From: Arun Radhakrishnan <x0051460@ti.com>
Date: Wed, 29 Jul 2009 13:25:38 -0500
Subject: [PATCH] SYSLINK: comment out mailbox clocks for OMAP4

This patch removes the access to the mailbox clocks since
the clock framework for OMAP4 is not ready.

Signed-off-by: Arun Radhakrishnan <x0051460@ti.com>
---
 arch/arm/mach-omap2/mailbox.c |   20 +++++++++++++-------
 1 files changed, 13 insertions(+), 7 deletions(-)
 mode change 100644 => 100755 arch/arm/mach-omap2/mailbox.c

diff --git a/arch/arm/mach-omap2/mailbox.c b/arch/arm/mach-omap2/mailbox.c
old mode 100644
new mode 100755
index 8b4e0e8..c506f97
--- a/arch/arm/mach-omap2/mailbox.c
+++ b/arch/arm/mach-omap2/mailbox.c
@@ -89,12 +89,14 @@ static int omap2_mbox_startup(struct omap_mbox *mbox)
 {
 	unsigned int l;
 
-	mbox_ick_handle = clk_get(NULL, "mailboxes_ick");
-	if (IS_ERR(mbox_ick_handle)) {
-		printk(KERN_ERR "Could not get mailboxes_ick\n");
-		return -ENODEV;
-	}
+	if (!cpu_is_omap44xx()) {
+		mbox_ick_handle = clk_get(NULL, "mailboxes_ick");
+		if (IS_ERR(mbox_ick_handle)) {
+			printk(KERN_ERR "Could not get mailboxes_ick\n");
+			return -ENODEV;
+		}
 	clk_enable(mbox_ick_handle);
+	}
 
 	l = mbox_read_reg(MAILBOX_REVISION);
 	pr_info("omap mailbox rev %d.%d\n", (l & 0xf0) >> 4, (l & 0x0f));
@@ -111,8 +113,12 @@ static int omap2_mbox_startup(struct omap_mbox *mbox)
 
 static void omap2_mbox_shutdown(struct omap_mbox *mbox)
 {
-	clk_disable(mbox_ick_handle);
-	clk_put(mbox_ick_handle);
+	if (!cpu_is_omap44xx()) {
+		clk_disable(mbox_ick_handle);
+		clk_put(mbox_ick_handle);
+	} else {
+	printk(KERN_ERR "OMAP4 clocks are not modeled");
+	}
 }
 
 /* Mailbox FIFO handle functions */
-- 
1.5.4.3

